### PR TITLE
Added Bin Sort in Typescript

### LINKED ISSUE #436: Bin-Sort in Typescript
This is an open issue for bin sorting algorithm implementation in Typescript. If you wish to contribute, just make some noise and we will assign you.

### PR DESCRIPTION
**Pull request by BLasan**
**Fixes #436**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    for(let element of arrayList){
+        element.sort()
+    }
+
+    var index: number = 0
+    for(let i=0; i<size; i++){
+        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
+            collection[index++] = arrayList[i][j]
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:22]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        arrayList[i] = new Array()
+    }
+
+    for(let i=0; i<size; i++){
+        var index = Math.floor(collection[i]) * size
+        arrayList[index].push(collection[i])
+    }
+
+    for(let element of arrayList){
+        element.sort()
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:16]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:1]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    for(let i=0; i<size; i++){
+        var index = Math.floor(collection[i]) * size
+        arrayList[index].push(collection[i])
+    }
+
+    for(let element of arrayList){
+        element.sort()
+    }
+
+    var index: number = 0
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:19]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            collection[index++] = arrayList[i][j]
+        }
+    }
+
+    for(let element of collection){
+        console.log(element)
+    }
+}
+
+var list: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:31]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 4 above)
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:19]: Codacy found an issue: ['index' is already defined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+    var index: number = 0
+    for(let i=0; i<size; i++){
+        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
+            collection[index++] = arrayList[i][j]
+        }
+    }
+
+    for(let element of collection){
+        console.log(element)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:27]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    var arrayList: number[][] = new Array<number>(size).map(() => new Array());
+
+    for(let i=0; i<size; i++){
+        arrayList[i] = new Array()
+    }
+
+    for(let i=0; i<size; i++){
+        var index = Math.floor(collection[i]) * size
+        arrayList[index].push(collection[i])
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:12]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
+function binSort(collection: Array<number>, size:number): void {
+    if(size <= 0) return;
+
+    var arrayList: number[][] = new Array<number>(size).map(() => new Array());
+
+    for(let i=0; i<size; i++){
+        arrayList[i] = new Array()
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:7]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
+function binSort(collection: Array<number>, size:number): void {
+    if(size <= 0) return;
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:2]: Codacy found an issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 4 above)
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:19]: Codacy found an issue: ['index' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+        }
+    }
+
+    for(let element of collection){
+        console.log(element)
+    }
+}
+
+var list: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434)
+binSort(list, list.length)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:32]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:22]: Codacy found an issue: ['index' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    if(size <= 0) return;
+
+    var arrayList: number[][] = new Array<number>(size).map(() => new Array());
+
+    for(let i=0; i<size; i++){
+        arrayList[i] = new Array()
+    }
+
+    for(let i=0; i<size; i++){
+        var index = Math.floor(collection[i]) * size
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:11]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 6 above)
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:27]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    for(let i=0, index=0; i<size; i++){
+        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
+            collection[index++] = arrayList[i][j];
+        }
+    }
+
+    for(let element of collection){
+        console.log(element);
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:28]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 13]
```diff
@@ -0,0 +1,33 @@
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:1]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:1]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
+            collection[index++] = arrayList[i][j];
+        }
+    }
+}
+
+var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
+binSort(collection, collection.length);
+for(let element of collection){
+    console.log(element);
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:31]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 15]
```diff
@@ -0,0 +1,29 @@
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:1]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:1]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 16]
```diff
@@ -0,0 +1,29 @@
+var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
+var arrayList: number[][] = new Array<Array<number>>(collection.length)
+
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:4]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 17]
```diff
@@ -0,0 +1,29 @@
+var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
+var arrayList: number[][] = new Array<Array<number>>(collection.length)
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:2]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 18]
```diff
@@ -0,0 +1,29 @@
+var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
+var arrayList: number[][] = new Array<Array<number>>(collection.length);
+
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:4]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

**Comment by BLasan** [comment]
@Danushka96 Please review

File: algorithms/ar-binsrt/typescript/bin_sort.ts (hunk 18 above)
Comment (BLasan) [reply algorithms/ar-binsrt/typescript/bin_sort.ts:4]: Can you help me out to fix this?

File: algorithms/ar-binsrt/typescript/bin_sort.ts [hunk 19]
```diff
@@ -0,0 +1,23 @@
+var collection: Array<number> = new Array(0.897, 0.565, 0.656, 0.1234, 0.665, 0.3434);
+var arrayList: number[][] = new Array<Array<number>>(collection.length);
+function binSort(collection: Array<number>, size:number): void {
```
Comment (RavinduSachintha) [inline algorithms/ar-binsrt/typescript/bin_sort.ts:3]: Codacy found an issue: [Function 'binSort' has a complexity of 8. Maximum allowed is 6.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)

**Comment by BLasan** [comment]
@Danushka96 Please review this

**Review by Danushka96 — APPROVED** [review]

**Pull request merged**